fix: allow multiple addressed in replyTo

mastermunj opened the pull request:
<!-- CLICK "Preview" FOR INSTRUCTIONS IN A MORE READABLE FORMAT -->

## Proposed changes

As of now, `replyTo` accepts only a single address. On the next `replyTo` call, the previous value is overridden.

## Types of changes

What types of changes does your code introduce?

_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/adonisjs/mail/blob/master/.github/CONTRIBUTING.md) doc
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

mastermunj commented:
Hi @thetutlage, please help review the PR.

Julien-R44 commented:
It's a breaking change. I think we'd better define a new function, or better, make `replyTo` accept either a string or an array of strings. like nodemailer

mastermunj commented:
The usage doesn't change. It still allows adding a single replyTo or by chaining, adding multiple replyTo. It doesn't seem like a breaking change.

Julien-R44 commented:
Oh yeah indeed, sorry, I read too quickly. Forget what I just said

mastermunj commented:
Thank you. I noticed that the tests are failing. Is there anything I can do about it?

Julien-R44 commented:
No, don't worry, this is normal. Some tests need environment variables and they are not injected in workflows that are triggered by a pull request from a fork

mastermunj commented:
Understood. Is it possible to merge this PR and do a release sooner?

Julien-R44 commented:
I still think this is not a really great API to offer. It seems to be a workaround or a hackish way to add multiple addresses.

If I want to add several replyTo I have to do this : 
```
message.replyTo('foo@bar.com')
message.replyTo('foo2@bar.com')
message.replyTo('foo3@bar.com')
```

Doesn't look good

To me the best thing is, as I said above, a new function or overloading this one so that it also accepts an array. Preference for overloading

Julien-R44 commented:
Also, personally, when I call `replyTo` twice in a row, I would expect the replyTo to be totally replaced, not to have things added to. Anyway I find that a bit weird and not really intuitive

mastermunj commented:
@thetutlage Nodemailer supports multiple `replyTo`. It was added in version `6.7.8`.

The conversation and confirmation are available at https://github.com/nodemailer/nodemailer/issues/1237

mastermunj commented:
Hi Julien and Virk. Since nodemailer allows multiple `replyTo` similar to `to` and `cc`, will there be a need for an additional test case?

mastermunj commented:
@thetutlage Please review whether this PR can be merged.

mastermunj commented:
Hi @thetutlage, I request you to consider this PR if possible. Let me know if any changes are needed.

RomainLanz commented:
Hey @mastermunj! 👋🏻 

Could you please update the test suite for the multiple `replyTo` case?

mastermunj commented:
Hi @RomainLanz ,

The behavior of `replyTo` is the same as that of `cc`.

The test cases for `replyTo` also are thus, kept the same as that of `cc`.

RomainLanz commented:
It does not matter if the behavior is the same. We need a test to ensure adding multiple `replyTo` will push in the `nodeMailerMessage.replyTo` array and `message.toJSON().message` will send back an object with the added array in.

mastermunj commented:
Maybe I was not able to explain in my previous comment, however, I have modified the existing test case for `replyTo` to be the same as the test case for `cc`.

I have also added a test case for multiple replyTo.

Here's the PR for the documentation repo https://github.com/adonisjs/docs.adonisjs.com/pull/210

Please review and let me know if any other change is required.

RomainLanz commented:
Thanks!
